ci/ray_ci/bisect/bisect_test.py

```diff
@@ -11,8 +17,36 @@
 @click.argument("failing_commit", required=True, type=str)
 def main(test_name: str, passing_commit: str, failing_commit: str) -> None:
     ci_init()
-    blame = Bisector(test_name, passing_commit, failing_commit, MacOSValidator()).run()
-    logger.info(f"Blame revision: {blame}")
+    test_target = _get_test_target(test_name)
+    blame_commit = Bisector(
+        test_target,
+        passing_commit,
+        failing_commit,
+        MacOSValidator(),
+    ).run()
+    logger.info(f"Blame revision: {blame_commit}")
+    _update_test_state(Test.gen_from_name(test_name), blame_commit)
+
+
+def _get_test_target(test_name: str) -> str:
+    for prefix in TEST_PREFIXES:
+        if test_name.startswith(prefix):
+            return test_name[len(prefix) :]
+    return test_name
+
+
+def _update_test_state(test: Test, blamed_commit: str) -> None:
+    test.update_from_s3()
+    logger.info(f"Test object: {json.dumps(test)}")
+    test[Test.KEY_BISECT_BLAMED_COMMIT] = blamed_commit
+
+    # Compute and update the next test state, then comment blamed commit on github issue
+    sm = CITestStateMachine(test)
+    sm.move()
+    sm.comment_blamed_commit_on_github_issue()
+
+    logger.info(f"Test object: {json.dumps(test)}")
+    test.persist_to_s3()
```

Comment:
should this be updated immediately after `test[Test.KEY_BISECT_BLAMED_COMMIT]`, so that a failure on `comment_blamed_commit_on_github_issue()` will not block the state machine's movement?

Reply:
yes, let's do it right after the `sm.move()` (which updates the test state)